components/AddCredits.vue

```diff
@@ -155,23 +155,28 @@
 
 <script setup lang="ts">
 	import { readItems } from '@directus/sdk';
+	import { useUserFilter } from '~/composables/useUserFilter';
 	import { useAuth } from '~/store/auth';
 	import { useMetadata } from '~/store/metadata';
 	const { $directus } = useNuxtApp();
 
 	const auth = useAuth();
-	const { user } = storeToRefs(auth);
+	const { user, adminMode, impersonation } = storeToRefs(auth);
 	const metadata = useMetadata();
+	const { getUserFilter } = useUserFilter();
 
 	defineEmits([ 'cancel', 'adopt-a-probe' ]);
 
 	const { data: adoptionsExists } = await useLazyAsyncData('gp_adopted_probes_exist', async () => {
 		const adoptions = await $directus.request(readItems('gp_probes', {
-			filter: { userId: { _eq: user.value.id } },
+			filter: getUserFilter('userId'),
 			limit: 1,
 		}));
 		return !!adoptions.length;
-	}, { default: () => false });
+	}, {
+		default: () => false,
+		watch: [ adminMode, impersonation ],
+	});
```

Comment:
I think this adds lots of complexity for an admin-only feature, and even now, it doesn't work in some places. I'd say it's ok to simply do a full page reload when these change and remove the watchers everywhere.